hooks/check

```diff
@@ -58,29 +58,34 @@ else
   describe "  runtime config [#R{FAILED}]"
 fi
 
-# Environment Parameter checks
 env_ok=yes
-if want_feature 'tls'; then
-  vault="secret/$GENESIS_VAULT_PREFIX"
-
-  # Check if haproxy certs are in the right location - they moved in v1.9.0 to
-  # conform to the contract of where generated certificates are placed
-  if ! safe --quiet exists "$vault/haproxy/certs/ca" >/dev/null 2>&1 ; then
-    if safe --quiet exists "$vault/haproxy/ca" >/dev/null 2>&1 ; then
-      safe --quiet mv "$vault/haproxy/ca"  "$vault/haproxy/certs/ca" >/dev/null 2>&1
+vault="secret/$GENESIS_VAULT_PREFIX"
+secret_exists() {
+  safe --quiet exists "$vault/$1" >/dev/null 2>&1
+}
+relocate_secret() {
+  src="$1"; dst="$2";
+  if ! secret_exists "$dst" ; then
+    if secret_exists "$src" ; then
+      safe --quiet mv "$vault/$src" "$vault/$dst" >/dev/null 2>&1
     else
-      describe "Cannot find HAProxy CA Certificates - please run \`genesis add-secrets $GENESIS_ENVIRONMENT\`"
-      env_ok=no
-    fi
-  fi
-  if ! safe --quiet exists "$vault/haproxy/certs/ssl" >/dev/null 2>&1 ; then
-    if safe --quiet exists "$vault/haproxy/ssl" >/dev/null 2>&1 ; then
-      safe --quiet mv "$vault/haproxy/ssl"  "$vault/haproxy/certs/ssl" >/dev/null 2>&1
-    elif [[ $env_ok == "yes" ]] ; then
-      describe "Cannot find HAProxy SSL Certificates - please run \`genesis add-secrets $GENESIS_ENVIRONMENT\`"
       env_ok=no
+      return 1
     fi
+  elif secret_exists "$src" ; then
+    # Remove outdated location
+    safe --quiet rm "$vault/$src" >/dev/null 2>&1:
   fi
+  return 0
+}
+
+# Environment Parameter checks
+if want_feature 'tls'; then
+
+  # Check if haproxy certs are in the right location - they moved in v1.9.0 to
+  # conform to the contract of where generated certificates are placed
+  relocate_secret "haproxy/ca"  "haproxy/certs/ca"  || describe "Cannot find HAProxy CA Certificates - please run \`genesis add-secrets $GENESIS_ENVIRONMENT\`"
+  relocate_secret "haproxy/ssl" "haproxy/certs/ssl" || describe "Cannot find HAProxy SSL Certificates - please run \`genesis add-secrets $GENESIS_ENVIRONMENT\`"
```

Comment:
These error messages are misleading when the provided_cert feature is enabled.

Reply:
There actually isn't a `provided_cert` feature, but there is an absence of a `self-signed` feature, and your point is valid for that case.  In light of that, I have rethought changing the location of this cert and will revert those changes (as the location is the same if it is provided or generated) .